presto-phoenix/src/main/java/io/prestosql/plugin/phoenix/PhoenixSplit.java

```diff
@@ -0,0 +1,184 @@
+        this.schemaName = schemaName;
+        this.tableName = requireNonNull(tableName, "table name is null");
+        this.tupleDomain = requireNonNull(tupleDomain, "tupleDomain is null");
+        this.addresses = addresses;
+        this.phoenixInputSplit = wrappedPhoenixInputSplit;
+    }
+
+    @JsonProperty
+    public String getConnectorId()
+    {
+        return connectorId;
+    }
+
+    @JsonProperty
+    @Nullable
+    public String getCatalogName()
+    {
+        return catalogName;
+    }
+
+    @JsonProperty
+    @Nullable
+    public String getSchemaName()
+    {
+        return schemaName;
+    }
+
+    @JsonProperty("phoenixInputSplit")
+    public WrappedPhoenixInputSplit getWrappedPhoenixInputSplit()
+    {
+        return phoenixInputSplit;
+    }
+
+    @JsonIgnore
+    public PhoenixInputSplit getPhoenixInputSplit()
+    {
+        return phoenixInputSplit.getPhoenixInputSplit();
+    }
+
+    @JsonProperty
+    public String getTableName()
+    {
+        return tableName;
+    }
+
+    @JsonProperty
+    public TupleDomain<ColumnHandle> getTupleDomain()
+    {
+        return tupleDomain;
+    }
+
+    @JsonProperty
+    @Override
+    public List<HostAddress> getAddresses()
+    {
+        return addresses;
+    }
+
+    @Override
+    public boolean isRemotelyAccessible()
+    {
+        return true;
+    }
+
+    @Override
+    public Object getInfo()
+    {
+        return ImmutableMap.builder()
+                .put("hosts", addresses)
+                .put("schema", schemaName)
+                .put("table", tableName)
+                .put("keyRange", getPhoenixInputSplit().getKeyRange())
+                .build();
+    }
+
+    @Override
+    public int hashCode()
+    {
+        return Objects.hash(
+                connectorId,
+                catalogName,
+                schemaName,
+                tableName,
+                tupleDomain,
+                addresses,
+                phoenixInputSplit);
+    }
+
+    @Override
+    public boolean equals(Object obj)
+    {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null || getClass() != obj.getClass()) {
+            return false;
+        }
+        PhoenixSplit other = (PhoenixSplit) obj;
+        return Objects.equals(this.connectorId, other.connectorId) &&
+                Objects.equals(this.catalogName, other.catalogName) &&
+                Objects.equals(this.schemaName, other.schemaName) &&
+                Objects.equals(this.tableName, other.tableName) &&
+                Objects.equals(this.tupleDomain, other.tupleDomain) &&
+                Objects.equals(this.addresses, other.addresses) &&
+                Objects.equals(this.phoenixInputSplit, other.phoenixInputSplit);
+    }
+
+    @Override
+    public String toString()
+    {
+        ToStringHelper helper = toStringHelper(this)
+                .add("connectorId", connectorId);
+        if (catalogName != null) {
```

Comment:
Rather than checking for nulls, add them unconditionally and call `omitNullValues()` at the end.